numba/cpython/unicode.py

```diff
@@ -842,6 +831,58 @@ def _adjust_indices(length, start, end):
     return start, end
 
 
+@overload_method(types.UnicodeType, 'startswith')
+def unicode_startswith(s, prefix, start=None, end=None):
+    if not is_nonelike(start) and not isinstance(start, types.Integer):
+        raise TypingError(
+            "When specified, the arg 'start' must be an Integer or nonelike")
+
+    if not is_nonelike(end) and not isinstance(end, types.Integer):
+        raise TypingError(
+            "When specified, the arg 'end' must be an Integer or nonelike")
+
+    if isinstance(prefix, (types.Tuple, types.UniTuple)):
+        if isinstance(prefix, types.Tuple) or \
+                not isinstance(prefix.dtype, types.UnicodeType):
+            raise TypingError(
+                "The arg 'prefix' should a UniTuple of UnicodeType")
+
+        def startswith_tuple_impl(s, prefix, start=None, end=None):
+            for item in prefix:
+                if s.startswith(item, start, end):
+                    return True
+            return False
+
+        return startswith_tuple_impl
+
+    if isinstance(prefix, types.UnicodeCharSeq):
+        def startswith_char_seq_impl(s, prefix, start=None, end=None):
+            return s.startswith(str(prefix), start, end)
+
+        return startswith_char_seq_impl
+
+    if isinstance(prefix, types.UnicodeType):
+        def startswith_unicode_impl(s, prefix, start=None, end=None):
+            length, prefix_length = len(s), len(prefix)
+            if start is None:
+                start = 0
+            if end is None:
+                end = length
+
+            start, end = _adjust_indices(length, start, end)
+            if end - start < prefix_length:
+                return False
+
+            if prefix_length == 0:
+                return True
+
+            s_slice = s[start:end]
+
+            return _cmp_region(s_slice, 0, prefix, 0, prefix_length) == 0
+
+        return startswith_unicode_impl
+
```

Comment:
I think there should be an `else` here for the case where `prefix` is of an unsupported type, which raises:
```python
    else:
        raise TypingError("The arg 'prefix' should be a string or a tuple of strings")
```